### PR TITLE
Williamhe/1546176 project spotlight golang

### PR DESCRIPTION
**Pull request by william-msft**
- [x] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1546176
- [x] Tests are included and/or updated for code changes.
- [x] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/Golang/GolangBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -23,7 +23,11 @@ echo "   "
 
 # TODO: add support for nested dirs
 echo "building go app..."
-go build -o oryxBuildBinary
+doc="https://aka.ms/troubleshoot-go"
+suggestion="Please check your go.mod is valid. Try building locally first: go build"
```
Comment (cormacpayne) [inline src/BuildScriptGenerator/Golang/GolangBashBuildSnippet.sh.tpl:27]: nit: I would slightly rephrase the last sentence here:

_Try building locally first with the following command: go build_

File: images/build/logger.sh [hunk 2]
```diff
@@ -25,7 +25,7 @@ function LogErrorWithTryCatch()
 
     # try
     set +e
-    output=$( $cmd )
+    output=$( $cmd 2>&1)
```
Comment (cormacpayne) [inline images/build/logger.sh:28]: Out of curiosity, what is this change needed for?

**Review by cormacpayne — DISMISSED** [review]

File: src/BuildScriptGenerator/Golang/GolangBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (william-msft) [reply src/BuildScriptGenerator/Golang/GolangBashBuildSnippet.sh.tpl]: I updated, thanks!

File: images/build/logger.sh (hunk 2 above)
Comment (william-msft) [reply images/build/logger.sh:28]: The `2>&1` will redirect the STDERR to STDOUT of the `$cmd`. Which is very important for capturing the error message of the command. Without it, only STDOUT gets captured in the `output`. Great question!

Comment (william-msft) [reply images/build/logger.sh]: I updated that line with a comment 😄

**Review by daniv-msft — APPROVED** [review]
Looks good to me
